app/tasks/raster_tile_cache_assets/symbology.py

```diff
@@ -51,67 +51,44 @@ class SymbologyInfo(NamedTuple):
     function: SymbologyFuncType
 
 
-def generate_date_conf_calc_string() -> str:
+def date_conf_merge_calc() -> str:
     """Create the calc string for classic GLAD/RADD alerts."""
-    day = "(A - ((A >= 30000) * 10000) - ((A >= 20000) * 20000))"
-    confidence = "(1 * (A >= 30000))"  # 0 for low confidence, 1 for high
+    day = "(A >= 20000) * (A.data % 10000)"
+    confidence = "(A.data // 30000)"  # 0 for low confidence, 1 for high
 
     red = f"({day} / 255)"
     green = f"({day} % 255)"
-    blue = f"(({confidence} + 1) * 100 + B)"
+    blue = f"(A.data >= 20000) * (({confidence} + 1) * 100 + B.data)"
 
     return f"np.ma.array([{red}, {green}, {blue}])"
 
 
-def generate_8_bit_integrated_calc_string() -> str:
+def integrated_alert_merge_calc() -> str:
     """Create the calc string needed to encode/merge the 8-bit integrated alerts
     :return:
     """
     # <LONG EXPLANATION WITH EXAMPLE CODE>
 
-    _first_alert = """
-    np.ma.array(
-        np.ma.array(
-            np.minimum.reduce([A.filled(65535), B.filled(65535), C.filled(65535)]),
-            mask=(A.mask & B.mask & C.mask)
-        ).filled(0),
-        mask=(A.mask & B.mask & C.mask)
-    )
-    """
-    first_alert = "".join(_first_alert.split())
-
-    first_day = f"({first_alert} >> 1)"
-
-    # At this point confidence is encoded as 0 for high, 1 for low.
-    # Reverse that here for use in the Blue channel
-    first_confidence = f"(({first_day} > 0) * (({first_alert} & 1) == 0) * 1)"
+    # For dateconf v1.5
+    day = "(A.data % 10000)"
+    confidence = "(A.data // 30000)"  # 0 for low confidence, 1 for high
 
-    intensity = "D.data"
-
-    red = f"({first_day} / 255).astype(np.uint8)"
-    green = f"({first_day} % 255).astype(np.uint8)"
-    blue = (
-        f"(({first_day} > 0) * "
-        f"(({first_confidence} + 1) * 100 + {intensity}))"
-        ".astype(np.uint8)"
-    )
+    red = f"({day} / 255)"
+    green = f"({day} % 255)"
+    blue = f"(A.data >= 20000) * (({confidence} + 1) * 100 + B.data)"
 
-    gladl_conf = (
-        "((A.filled(0) >> 1) > 0) * "
-        "(((A.filled(0) & 1) == 0) * 2 + (A.filled(0) & 1))"
-    )
-    glads2_conf = (
-        "((B.filled(0) >> 1) > 0) * "
-        "(((B.filled(0) & 1) == 0) * 2 + (B.filled(0) & 1))"
-    )
-    radd_conf = (
-        "((C.filled(0) >> 1) > 0) * "
-        "(((C.filled(0) & 1) == 0) * 2 + (C.filled(0) & 1))"
+    # The front end is also expecting the confidences of all the alerts in the
+    # original alert systems, packed into a uint16 as the Alpha channel.
+    # Fake the combined confidences because the front end doesn't need to know
+    # which alert systems are what confidence. It REALLY wants to know if the
+    # combined  alert is low, high, or highest confidence. In the future
+    # suggest a new way for the front end to obtain this info that's more
+    # elegant (such as by more efficiently packing into Blue channel)
+    alpha = (
```

Comment:
Just to understand, you're basically just putting the combined confidence in the alpha  band, and where you'd normally have the bits for the other confidences, it's just empty?

Reply:
Yes! With one other detail that isn't clear because I made a Math error: For the highest confidence case it ALSO indicates an alert in one of the other bits (I arbitrarily chose RADD for a single alert and RADD + GLADS2 for multiple). The frontend just checks "Is there an alert? In more than one system? If so: Highest. If not: Confidence is the one indicated in that one system."
I will push a better explanation and fix the Math error.